pilot/docker/Dockerfile.proxy_debug

```diff
@@ -1,40 +1,15 @@
-FROM ubuntu:xenial
-# Do not add more stuff to this list that isn't small or critically useful.
-# If you occasionally need something on the container do
-# sudo apt-get update && apt-get whichever
-RUN apt-get update && \
-    apt-get install --no-install-recommends -y \
-      curl \
-      iptables \
-      iproute2 \
-      iputils-ping \
-      dnsutils \
-      netcat \
-      tcpdump \
-      net-tools \
-      libc6-dbg gdb \
-      elvis-tiny \
-      lsof \
-      busybox \
-      sudo && \
-    rm -rf /var/lib/apt/lists/*
+FROM istionightly/base_debug
```

Comment:
@costinm why are we re-introducing a dependency on another Dockerfile? This was the source of a lot of confusion previously.  If we wanted to try out local changes to the base setup, it would seem that relying on an external image would be troublesome.

Reply:
It takes very long to re-build the base image - I want it to be built by the nightly job.

Also I want to be able to optionally use the debug base image for the pilot test app ( and possibly pilot, mixer ).

Reply:
ok, that seems reasonable.  To be clear: there is no istio-specific stuff going on in the base image, correct?

Reply:
There is some istio-specific stuff right now ( I can move it back ) - there is an istio-proxy user created. I don't think it hurts in any way to have the user - but it'll help for some tests and 
use cases. 

The base image is just a refactoring of what we had - to avoid 'apt-get update /install' in each
build and to allow some reuse.

Reply:
@costinm Yeah, let's move it back.  Having istio-specific stuff stuck in a base image that isn't part of the local build is troublesome for development.

Reply:
Done.